### PR TITLE
More wiimote controls (+extensions) for android

### PR DESCRIPTION
**Pull request by Phatcat**
Should hopefully add some more functionality to using wiimote controls
on android.

### CODE REVIEW

**Comment by degasus** [comment]
@sigmabeta Do you want to review?

File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/NativeLibrary.java [hunk 1]
```diff
@@ -25,39 +25,56 @@
...
-		public static final int WIIMOTE_BUTTON_MINUS = 24;
-		public static final int WIIMOTE_BUTTON_PLUS  = 25;
-		public static final int WIIMOTE_BUTTON_HOME  = 26;
-		public static final int WIIMOTE_BUTTON_1     = 27;
-		public static final int WIIMOTE_BUTTON_2     = 28;
-		public static final int WIIMOTE_UP           = 29;
-		public static final int WIIMOTE_DOWN         = 30;
-		public static final int WIIMOTE_LEFT         = 31;
-		public static final int WIIMOTE_RIGHT        = 32;
+		public static final int BUTTON_A              = 0;
```
Comment (bb010g) [inline Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/NativeLibrary.java]: Do we want these aligned? It just causes issues like this in diffs.

Comment (Phatcat) [inline Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/NativeLibrary.java]: I just thought to stick to formatting as it currently was, but I see what you mean. It's only the 1 file, though, and not too many lines.

Comment (JosJuice) [inline Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/NativeLibrary.java]: The formatting matters more than the diff, in my opinion. If you want to view the diff without whitespace changes, you can use the URL https://github.com/dolphin-emu/dolphin/pull/3253/files?w=1

**Comment by sigmabeta** [comment]
The android framework side of this looks mostly good. (the xml) I did not check thoroughly to make sure all number incrementations happened correctly as I am on mobile for now. But it looks fine. 

I'll have another look later, and maybe @hdkr should also.

**Comment by sigmabeta** [comment]
Uh @sonicadvance1 that is.

**Comment by sigmabeta** [comment]
I haven't had a chance to try this out myself, but if anyone else has and can vouch that it works, I don't see any reason not to merge.

**Comment by sigmabeta** [comment]
@dolphin-emu-bot rebuild

**Comment by sigmabeta** [comment]
I haven't had a chance to try this out myself, but if anyone else has and can vouch that it works, I don't see any reason not to merge once the build failures / lint checks are fixed

@Phatcat, once you've done that, ping me and I'll have buildbot run the checks again.

**Comment by Phatcat** [comment]
I have debugged the program and locally it now builds. One problem, though, the extensions don't seem to respond, that is; adding the line 'Extension = Classic' to the wiimote.ini file will cause games, like for example SSBB, to not react to any of the normal wiimote buttons, nor any of the extension buttons; yet the home button will work on the wiimote, and inside the home menu, the wiimote buttons will respond as normally. It would seem that the emulator recognizes the extension and maybe even accepts it, since it suspends some wiimote buttons, but any extension button that is pressed has no effect whatsoever, nomatter the extensions. 
Either I remove anything extension related from this pull request and then full standalone wiimote should be working for android and ready to merge, or I keep working on getting extensions to work and wait to merge till everything is working.

**Comment by MaJoRoesch** [comment]
@lioncash Do you want to review this?

**Comment by JosJuice** [comment]
@dolphin-emu-bot rebuild

**Comment by degasus** [comment]
@sigmabeta ping :D

**Comment by Phatcat** [comment]
Tested this on my Nvidia Shield and seems to work fine, all controls responding in smash bros brawl.
There might be some issues if you have the same physical button mapped to 2 different controls in-game, also, as for now, the option to select extension in the configurations is purely cosmetic; it does nothing as of yet. I am not really sure as of how to implement this, letting the user configure the extension from within the app, so for now, actually setting/changing the extension has to be done manually outside the app in the WiimoteNew.ini file.

Other than that, Nunchuk and Classic controller buttons have been tested and found working on the Nvidia Shield Android TV (at least inside smash bros brawl). I do not own any 'band' or 'hero' games, so I myself can't test the instrument extensions.
This needs testing on touch devices as well.

**Comment by degasus** [comment]
@awickham Do you want to also test this PR?

**Comment by Phatcat** [comment]
Alright, so now the extension should be configurable from within the app - there is no reason to specifically test this on touch devices as it adds nothing to the input overlay. Currently, there are no wiimote controls at all in the overlay input. It might require a slight rewrite of the inputoverlay, so this pr is all controller-input. 

I would still like for someone besides myself to test this as well, of course.

**Comment by JosJuice** [comment]
@dolphin-emu-bot rebuild

**Pull request merged**